src/database.rs

```diff
@@ -0,0 +1,63 @@
+extern crate rusqlite;
+
+pub struct StoredScript {
+    pub id: Option<i64>,
+    pub name: String,
+    pub source: String,
+    pub is_active: bool
+}
+
+pub struct ScriptDatabase {
+    connection: rusqlite::Connection
+}
+
+impl ScriptDatabase {
+
+    pub fn new(filename: String) -> Self {
+        let connection = rusqlite::Connection::open(filename).unwrap();
+        connection.execute("CREATE TABLE IF NOT EXISTS scripts (
+            id          INTEGER PRIMARY KEY AUTOINCREMENT,
+            name        TEXT NOT NULL UNIQUE,
+            source      TEXT NOT NULL,
+            is_active   BOOL NOT NULL DEFAULT 1
+        )", &[]).unwrap();
+
+        ScriptDatabase {
+            connection: connection
+        }
+    }
+
+    pub fn save(&self, script: &mut StoredScript) -> rusqlite::Result<()> {
```

Comment:
I'm not a big fan of calling this `save`. I initially suggested `put`, to  match REST style, but `insert` is actually probably better, to match the signatures of `HashSet`.